wsidicomizer/common.py

```diff
@@ -109,7 +113,9 @@ def create_instance_dataset(
                 DSfloat(self.pixel_spacing.height, True)
             ]
             pixel_measure_sequence.SpacingBetweenSlices = 0.0
-            pixel_measure_sequence.SliceThickness = 0.0
+            # DICOM 2022a part 3 IODs - C.8.12.4.1.2 Imaged Volume Width, Height, Depth
+            # Depth must not be 0. Default to 0.5 microns
+            pixel_measure_sequence.SliceThickness = 0.0005
```

Comment:
In my use case, I would supply a small number of fields via JSON. This would cause modules to not be None in dataset.py create_base_dataset() and create_default_modules() would never be called and thus many required fields would not be filled in. As a user I would hope that the conversion would supply all required fields and I could supply overrides only for things I needed to change instead of doing all the work of create_default_modules() myself. I may not have understood your design intentions correctly and have been keeping changes as small as possible to avoid affecting the overall design.

A proper way to set this to a real value is definitely needed.

Reply:
I think the use case that you describe is what we want to archive. Metadata could come from the user, from the image file, or (as a last resort for values that needs to be defined) by some default value. This is however not yet implemented.